src/main/java/MockClass.java

```diff
@@ -1,7 +1,21 @@
+import java.util.Optional;
+
 public class MockClass {
 
     public int doSomething(int seed) {
 
         return seed + 5;
     }
+
+    public void doSomething () {
+        Optional<String> optional = getOptional();
+        if (optional != null) {  // Noncompliant
+            // do something with optional...
+        }
+    }
+
+    public Optional<String> getOptional() {
+        // ...
+        return null;  // Noncompliant
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Methods with an "Optional" return type should never return null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.tiolktrack.byznass.com:9000/coding_rules#rule_key=squid%3AS2789)